_inc/lib/class-jetpack-instagram-gallery-helper.php

```diff
@@ -72,7 +72,10 @@ public static function get_instagram_gallery( $access_token, $count ) {
 		$cached_gallery = get_transient( $transient_key );
 		if ( $cached_gallery ) {
 			$decoded_cached_gallery = json_decode( $cached_gallery );
-			$cached_count           = count( $decoded_cached_gallery->images );
+			// `images` can be an array of images or a string 'ERROR'.
+			// 'ERROR' seems to be cached to limit the number of repeated API requests when there's a problem
+			// retrieving images.
+			$cached_count = is_array( $decoded_cached_gallery->images ) ? count( $decoded_cached_gallery->images ) : 0;
```

Comment:
I think this will mean we'll make the request every time when `$decoded_cached_gallery->images` equals `ERROR`. I'm not sure if that matters or not, because we'll be setting the transient on WPCOM too.

We could change the `if` statement logic to:

```
if ( 'ERROR' === $decoded_cached_gallery->images || count( $decoded_cached_gallery->images ) >= $count ) {
```

Although that does assume that if the value isn't `ERROR` then it's an array/countable. It should be, but you never know!

Reply:
I agree something is off here -- I don't understand the value of returning and storing `'ERROR'` rather than using a legitimate WP_Error in the API and a proper http status code. But I'm not going to address those in this PR.

Reply:
I agree, let's just stop the error for now.